File: rental-management/src/types/payment.ts
export interface Payment {
    id: string;
    tenantId: string;
    amount: number;
    dueDate: string;
    status: 'paid' | 'overdue' | 'pending';
    createdAt: string;
}

export interface PaymentHistory {
    tenantId: string;
    payments: Payment[];
}

File: RentEazy/app/src/main/java/com/renteazy/models/Payment.kt
package com.renteazy.models

data class Payment(
    val id: String = "",
    val propertyId: String = "",
    val tenantId: String = "",
    val ownerId: String = "",
    val amount: Double = 0.0,
    val type: String = "RENT", // RENT, SECURITY_DEPOSIT, UTILITY
    val status: String = "PENDING", // PENDING, COMPLETED, FAILED
    val dueDate: Long = 0,
    val paidDate: Long? = null,
    val month: Int = 0, // 1-12
    val year: Int = 0,
    val utilityDetails: Map<String, Double> = mapOf(), // e.g., "electricity" -> 1000.0
    val paymentMethod: String = "", // UPI, BANK_TRANSFER, etc.
    val transactionId: String = "",
    val notes: String = "",
    val createdAt: Long = System.currentTimeMillis()
)